backend/PythonClient/multirotor/monitor/point_deviation_monitor.py

```diff
@@ -125,21 +128,29 @@ def calculate_actual_distance(self):
 
     def draw_trace_3d(self):
         graph_dir = self.get_graph_dir()
+
         if not self.breach_flag:
             title = f"{self.target_drone} Planned vs. Actual\nDrone speed: {self.mission.speed} m/s\nWind: {self.wind_speed_text}"
         else:
             title = f"(FAILED) {self.target_drone} Planned vs. Actual\nDrone speed: {self.mission.speed} m/s\nWind: {self.wind_speed_text}"
-        grapher = ThreeDimensionalGrapher()
-        grapher.draw_trace_vs_planned(planed_position_list=self.mission.points,
+
+        self.grapher.draw_trace_vs_planned(planed_position_list=self.mission.points,
                                       actual_position_list=self.est_position_array,
                                       full_target_directory=graph_dir,
                                       drone_name=self.target_drone,
-                                      title=title
-                                      )
+                                      title=title)
 
-        grapher.draw_interactive_trace_vs_planned(planed_position_list=self.mission.points,
+        interactive_html_content = self.grapher.draw_interactive_trace_vs_planned(planed_position_list=self.mission.points,
                                                   actual_position_list=self.est_position_array,
                                                   full_target_directory=graph_dir,
                                                   drone_name=self.target_drone,
                                                   title=title
                                                   )
+        if interactive_html_content:
+            # Upload the HTML file directly to GC
+            file_name = f"html_reports/{self.target_drone}/interactive/{self.target_drone}_interactive.html"
+            self.air_sim_app.upload_to_gcs(file_name, interactive_html_content)
```

Comment:
This line might be what's causing the errors when uploading the HTML files to the GCS. Try seeing through brute force methods if you still get the error or not and then editing the pathway of the file upload. Overall, there may be a better way to upload the reports through a different path.